### ISSUE TITLE
Matplotlib as an optional dependency 

### ISSUE THREAD

**Issue by cheginit** [gboeing/osmnx#905]
**Do not delete the following template. Fill it out completely.**

  - [x] I have read the OSMnx [contributing guidelines](../blob/main/CONTRIBUTING.md)
  - [x] I have read the OSMnx [documentation](https://osmnx.readthedocs.io/) and [usage examples](https://github.com/gboeing/osmnx-examples)
  - [x] I have searched the open/closed issues and pull requests for others similar to mine

**Is your feature proposal related to a problem?**

Since `matplotlib` 3.6 is fairly recent, there are some packages that still are not compatible with it, for example [ProPlot](https://proplot.readthedocs.io). Therefore, since `osmnx` 1.3 sets the min version of `matplotlib` to 3.6, it might cause some issues with using it with other packages. 

**Describe the solution you'd like to propose**

I suggest to make `matplotlib` an optional dependency for the `plot` module, similar to `folium` for the `folium` module.

**Describe alternatives you've considered**

Another option can be setting the min version to of `matplotlib` to 3.5. I ran the test suite with this version, and they all passed. This is the env file that I used based on the one in the `test` directory:

```yaml
name: osmnx-dev
channels:
    - conda-forge
    - nodefaults
dependencies:
    # required
    - geopandas
    - networkx
    - numpy
    - pandas
    - pyproj
    - requests
    - shapely>=2.0

    # extras
    - matplotlib-base=3.5
    - folium
    - gdal
    - rasterio
    - scikit-learn
    - scipy

    # linting/testing
    - pre-commit
    - pytest-xdist
    - pytest-sugar
    - pytest-cov
    - sphinx
```

**Additional context**

If interested, I can submit a PR.

**Comment by gboeing** [comment]
Thanks for flagging this. Yes would you like to open a PR and downgrade matplotlib to 3.5? I had bumped that dependency version last week when I realized I was only testing against 3.6, but 3.5 should be just fine for a while longer.

**Comment by cheginit** [comment]
Sure.

**Comment by gboeing** [comment]
Closed by #907
